pkg/reconcile/cluster_reconciler.go

```diff
@@ -66,18 +75,93 @@ func canReconcileClusterBootstrapAgent(c *infrav3.Cluster) bool {
 	}
 }
 
+// DeleteForCluster delete bootstrap agent
+func (r *clusterReconciler) deleteBootstrapAgentForCluster(ctx context.Context, cluster *infrav3.Cluster) error {
```

Comment:
why are we seeing these changes in this pr? they are already merged to main and released right ?

Reply:
Yes you are correct. I understand this should not be part of this PR, but it came during rebase with the main. Now a new branch has come into place so we are merging only the required changes into it.